File: E_Kotlin/Lessons/G_Exception.kt

fun div(a:Int, b:Int):Int{
    if(b==0) throw IllegalArgumentException("Divisor cannot be 0")
    return a / b
}

fun main() {
    
    try{
        div(2,0)
        println("No should execute")
    }catch(e: IllegalArgumentException){
        println("Error: ${e.message}")
    }finally{
        println("Always executed")
    }
}

File: H_JavaScript/Lessons/G_Exception.js

try {
    let result = 10 / 0
    if(result==Infinity) throw new Error("Result cannot be Infinity, check operands")
    console.log(result)
} catch (error) {
    console.error("Error:", error.message);
} finally {
    console.log("Always execute.");
}
  
console.log("Ending...");